File: Spring Boot/microservices/distance-service/src/main/java/pl/edu/pg/benchmarking/distance/DistanceController.java
package pl.edu.pg.benchmarking.distance;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestBody;
import pl.edu.pg.benchmarking.distance.dto.DistanceRequest;
import pl.edu.pg.benchmarking.distance.dto.DistanceResponse;

import java.util.List;

@RestController
@RequestMapping("api/distance")
public class DistanceController {

    private final SphericalDistanceService distanceService;

    public DistanceController(SphericalDistanceService distanceService) {
        this.distanceService = distanceService;
    }

    @PostMapping
    public ResponseEntity<List<DistanceResponse>> calculateDistance(@RequestBody List<DistanceRequest> request) {
        List<Point> points = DistanceRequest.dtoToEntityMapper().apply(request);
        List<Double> distances = points.stream()
                .map(pointPair -> distanceService.calculateDistance(
                        pointPair.getLatitude1(), pointPair.getLongitude1(),
                        pointPair.getLatitude2(), pointPair.getLongitude2())
                ).toList();
        return ResponseEntity.ok(DistanceResponse.entityToDtoMapper().apply(points, distances));
    }
}

File: Dotnet/microservices/distance-service/Controllers/DistanceController.cs
using DistanceService.Requests;
using DistanceService.Responses;
using DistanceService.Services;
using DistanceService.Models;

using Microsoft.AspNetCore.Mvc;

namespace DistanceService.Controllers
{
    [ApiController]
    public class DistanceController : ControllerBase
    {
        private readonly SphericalDistanceService _sphericalDistanceService;

        public DistanceController(SphericalDistanceService sphericalDistanceService)
        {
            _sphericalDistanceService = sphericalDistanceService;
        }

        [Route("api/distance")]
        [HttpPost]
        public IActionResult Distance(List<DistanceRequest> req)
        {
            List<Point> points = DistanceRequest.DtoToEntityMapper(req);
            List<double> distances = points.Select(pointPair =>
                _sphericalDistanceService.CalculateDistance(
                    pointPair.Latitude1, pointPair.Longitude1, pointPair.Latitude2, pointPair.Longitude2)
            ).ToList();
            return Ok(DistanceResponse.EntityToDtoMapper(points, distances));
        }
    }
}
